File: stack/1544-Make-The-String-Great.go
func makeGood(s string) string {
    if len(s) <= 1 {
        return s
    }

    var stacks []rune
    for _, c := range s {
        if len(stacks) > 0 && c != stacks[len(stacks) - 1] && areEqualIgnoreCase(c, stacks[len(stacks) - 1]) {
            stacks = stacks[:len(stacks) - 1]
        } else {
            stacks = append(stacks, c)
        }
    }
    return string(stacks)
}

func areEqualIgnoreCase(a rune, b rune) bool {
    return unicode.ToLower(a) == unicode.ToLower(b)
}

File: stack/1544-Make-The-String-Great.py
class Solution:
    def makeGood(self, s: str) -> str:
        stacks = []

        for c in s:
            if stacks and stacks[-1] != c and stacks[-1].lower() == c.lower():
                stacks.pop()
            else:
                stacks.append(c)
        return "".join(stacks)